rabbitmq/queue_runner.go

```diff
@@ -0,0 +1,136 @@
+package rabbitmq
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	amqp "github.com/rabbitmq/amqp091-go"
+)
+
+const (
+	defaultPrefetch = 32
+	defaultType     = "quorum"
+)
+
+// runs a single queue with its own channel/consumer
+type queueRunner struct {
+	exchangeName string
+	queue        QueueSpec
+	ch           *amqp.Channel
+	chClose      chan *amqp.Error
+	consumerTag  string
+}
+
+func newQueueRunner(conn *amqp.Connection, exchangeName string, q QueueSpec) (*queueRunner, error) {
+	ch, err := conn.Channel()
+	if err != nil {
+		return nil, fmt.Errorf("failed to open channel for queue %s: %w", q.cfg.Name, err)
+	}
+
+	r := &queueRunner{
+		exchangeName: exchangeName,
+		queue:        q,
+		ch:           ch,
+		chClose:      ch.NotifyClose(make(chan *amqp.Error, 1)),
+		consumerTag:  fmt.Sprintf("cozy-%s-%s-%d", exchangeName, q.cfg.Name, time.Now().UnixNano()),
+	}
+
+	// Declare DLX/DLQ using resolved fields already set on QueueSpec
+	if q.dlxName != "" {
+		if err := r.ch.ExchangeDeclare(q.dlxName, "fanout", true, false, false, false, nil); err != nil {
+			_ = ch.Close()
+			return nil, fmt.Errorf("failed to declare DLX %s: %w", q.dlxName, err)
+		}
+	}
+	if q.dlqName != "" {
+		if _, err := r.ch.QueueDeclare(q.dlqName, true, false, false, false, nil); err != nil {
+			_ = ch.Close()
+			return nil, fmt.Errorf("failed to declare DLQ %s: %w", q.dlqName, err)
+		}
+		if q.dlxName != "" {
+			if err := r.ch.QueueBind(q.dlqName, "", q.dlxName, false, nil); err != nil {
+				_ = ch.Close()
+				return nil, fmt.Errorf("failed to bind DLQ %s to DLX %s: %w", q.dlqName, q.dlxName, err)
+			}
+		}
+	}
+
+	// Declare queue with args and bind to exchange
+	qArgs := amqp.Table{
+		"x-queue-type":           defaultType,
+		"x-dead-letter-exchange": q.dlxName,
+		"x-delivery-limit":       q.cfg.DeliveryLimit,
+	}
```

Comment:
Looks like we're missing 2 properties from the ADR:
- `x-dead-letter-strategy: at-least-once`
- `x-overflow: reject-publish`